proposed/event-dispatcher-meta.md

```diff
@@ -80,7 +80,37 @@ That left two possible options:
 * Allow events to be mutable.
 * Require, but be unable to enforce, immutable objects with a high-ceremony interface, more work for listener authors, and a higher potential for breakage that may not be detectable at compile time.
 
-Given those options the Working Group felt mutable events were the safer alternative.
+By "high-ceremony", we imply that verbose syntax and/or implementations would be required.  In the former case, listener authors would need to (a) create a new instance with the propagation flag toggled, and (b) return the new instance so that the dispatcher could examine it:
+
+```php
+function (SomeEvent $event) : SomeEvent
+{
+    // do some work
+    return $event->withPropagationStopped();
+}
+```
+
+In the latter case, dispatcher implementations, checks on the return value would be required:
+
+```php
+foreach ($provider->getListenersForEvent($event) as $listener) {
+    $returnedEvent = $listener($event);
+    
+    if (! $returnedEvent instanceof $event) {
+        // This is an exceptional case!
+    }
```

Comment:
The comment here should be expanded, as it's a key point.  The dispatcher has to handle this case because a buggy listener may not return an event, or the same event, or anything at all, all of which could break things in exciting ways.  Not relying on the return value of the listener removes a whole array of error conditions that no longer need to be addressed.

Reply:
I've pushed a change that should address this more fully.